src/server/player_sao.cpp

```diff
@@ -558,12 +558,40 @@ void PlayerSAO::setMaxSpeedOverride(const v3f &vel)
 
 bool PlayerSAO::checkMovementCheat()
 {
-	if (isAttached() || m_is_singleplayer ||
+	if (m_is_singleplayer ||
 			g_settings->getBool("disable_anticheat")) {
 		m_last_good_position = m_base_position;
 		return false;
 	}
 
+	if (isAttached()) {
+		int parent_id;
+		std::string bone;
+		v3f attachment_pos, attachment_rot;
+		getAttachment(&parent_id, &bone, &attachment_pos, &attachment_rot);
+		UnitSAO *parent = (UnitSAO *)getParent();
+		v3f pos = getBasePosition();
+		f32 radius = sqrt(attachment_pos.X * attachment_pos.X + attachment_pos.Z * attachment_pos.Z);
+		v3f parent_pos = parent->getBasePosition();
+		f32 parent_rot = parent->getRadYawDep() + acos(1 / radius);
+		attachment_pos.X = cos(parent_rot) * radius;
+		attachment_pos.Z = sin(parent_rot) * radius;
```

Comment:
The player can actually be attached to a bone of the parent which faces backwards. I think a simple range check here suffices - using `attachment_pos` as additional "tolerance zone".